app/grandchallenge/reader_studies/views.py

```diff
@@ -1053,19 +1053,17 @@ def destroy(self, request, *args, **kwargs):
             )
         return super().destroy(request, *args, **kwargs)
 
-    def filter_queryset(self, queryset):
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
         # Note: if more fields besides 'reader_study' are added to the
         # filter_set fields, we cannot call super anymore before randomizing
         # as we only want to filter out the display sets for a specific
         # reader study.
         reader_study = self.reader_study
-        queryset = super().filter_queryset(queryset)
         if reader_study and reader_study.shuffle_hanging_list:
-            set_seed(1 / int(self.request.user.pk))
-            queryset = queryset.order_by("?")
+            queryset = self.shuffle_qs(queryset=queryset)
             # Save the queryset to determine each item's index in the serializer
             self.randomized_qs = list(queryset)
```

Comment:
I'd say this should be part of the `shuffle_qs` method, as you do it after each call to that anyway.